bin/tx-prover/src/proxy/mod.rs

```diff
@@ -317,6 +322,16 @@ impl RequestContext {
     }
 }
 
+// LOAD BALANCER WRAPPER
+// ================================================================================================
+
+/// Wrapper around the load balancer that implements the ProxyHttp trait
+///
+/// This wrapper is used to implement the ProxyHttp trait for Arc<LoadBalancer>.
+/// This is necessary because we want to share the load balancer between the proxy server and the
+/// health check background service.
+pub struct LoadBalancerWrapper(pub Arc<LoadBalancer>);
```

Comment:
nit: Because now `ProxyHttp` is being implemented for the wrapper struct, I think we could rename `LoadBalancerWrapper` into `LoadBalancer` and `LoadBalancer` could be something like `LoadBalancerState` (or maybe `WorkerState`) since the idea of adding the wrapper was to be able to easily share the state

Reply:
I like `LoadBalancerState`, pretty much straightforward to wait it does. I'm renaming it.